### ISSUE TITLE
Leverage Microsoft.ApplicationInsights.WorkerService ?

### ISSUE THREAD

**Issue by czb182** [Azure/azure-webjobs-sdk#2394]
There is a somewhat new package Microsoft.ApplicationInsights.WorkerService that supports Generic Hosts.  I'm wondering if this package should now be used for WebJobs logging, or maybe it should be leveraged by Microsoft.Azure.WebJobs.Logging.ApplicationInsights in some way (or not).

**Comment by czb182** [comment]
Revisiting this for new webjob.  I haven't tried yet, but I'm wondering how well using the workservice AI package would work versus the package in this repo.  The AI package in this repo has dependences on asp.net which seems weird.

**Comment by v-anvari** [comment]
Hi @czb182 , Apologies for the delayed response. Kindly give us more details to understand this scenario better, what package are you using for your latest project and the usage of it your code. Can you give us the repro steps as well.

**Comment by czb182** [comment]
There is no repo, its just a question.  I've implemented webjobs sometimes using Microsoft.Azure.WebJobs.Logging.ApplicationInsights and sometimes using Microsoft.ApplicationInsights.WorkerService.  It seems like Microsoft.ApplicationInsights.WorkerService is a fit for webjobs, just wondering how it fits in.

**Comment by v-anvari** [comment]
Hi @czb182 , It is best suited for non-HTTP workloads like messaging, background tasks, console applications. Adding this package would allow you to use "Microsoft.ApplicationInsights;".

You can use either of the packages but it depends on the individual package dependencies 

 [Application Insights for Worker Service applications (non-HTTP applications)](https://docs.microsoft.com/en-us/azure/azure-monitor/app/worker-service)

Please feel free to reach out to us in case of any further queries regarding this issue

**Comment by dzendras** [comment]
I do think that the way AppInsights are handled in the current implementation is bad. WorkerService package is what WebJobs.SDK should rely on. Currently both packages have their own way of setting AI up. It turns out that the WebJobs.SDK way is flawed. 
There are numerous issues reporting that logging dependency telemetry stopped working. I came across it, too. As far as I am concerned, if you rely on WJ.SDK's AddApplicationInsightsWebJobs(...) then Dependency Telemetry is logged, but ONLY IF the dependencies are called from within WJ.SDK's own triggers (e.g. TimerTrigger). If your web job does something extra, e.g. exposes IHostedService which is running in an endless loop, then dependencies are NOT logged there.
In order to have all dependencies logged, you need to call both SDKs:
1. services.AddApplicationInsightsTelemetryWorkerService(...)
2. hostBuilder.ConfigureLogging((_, b) =>
            {
                b.AddApplicationInsightsWebJobs(...);
            });

If you change the order, then it does not work again. Unfortunately if you call it in the right order, then the resulting TelemetryConfiguration is a mess, contains duplicate modules, telemetry initializers etc.

To sum up, the current implementation is flaky, error-prone and does not follow DRY. Inspect AddApplicationInsightsWebJobs and you'll see that there is some code overlap with WorkerService 's implementation.
